libphal/exception.H

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include <exception>
+
+namespace libphal
+{
+namespace exception
+{
+
+enum ERR_TYPE {
+	NONE = 0,
+};
+
+typedef std::map<ERR_TYPE, const char *> errMsgMapType;
+const errMsgMapType errMsgMap = {{NONE, ""}};
+
+// libphal specific errors base exception class
+struct LibphalError : public std::exception {
+	virtual ERR_TYPE errType() const noexcept = 0;
+};
+
+} // namespace exception
+
+using libphalerror_t = exception::LibphalError;
+
```

Comment:
Any definitions within the namespace ideally should not have the same prefix repeated.   It can be simply `error_t`.